api/singlesignon.go

```diff
@@ -0,0 +1,116 @@
+package api
```

Comment:
nit: rename file name to `sso.go`?

Reply:
There's already a `sso.go` file that hosts the admin methods. But maybe I'll rename that to `ssoadmin.go` and this to `sso.go`. Great catch!